yb-voyager/src/tgtdb/suites/oracleSuite.go

```diff
@@ -0,0 +1,210 @@
+		}
+		if formatIfRequired {
+			formattedDate = fmt.Sprintf("TO_DATE('%s', 'DD-MM-YYYY')", formattedDate)
+		}
+		return formattedDate, nil
+	},
+	"io.debezium.time.Timestamp": func(columnValue string, formatIfRequired bool) (string, error) {
+		epochMilliSecs, err := strconv.ParseInt(columnValue, 10, 64)
+		if err != nil {
+			return columnValue, fmt.Errorf("parsing epoch milliseconds: %v", err)
+		}
+		epochSecs := epochMilliSecs / 1000
+		timestamp := time.Unix(epochSecs, 0).UTC()
+		oracleTimestampFormat := "02-01-2006 03.04.05.000 PM" //format: DD-MM-YY HH.MI.SS.FFF PM
+		formattedTimestamp := timestamp.Format(oracleTimestampFormat)
+		if formatIfRequired {
+			formattedTimestamp = fmt.Sprintf("TO_TIMESTAMP('%s','DD-MM-RR HH:MI:SS.FF9 AM')", formattedTimestamp)
+		}
+		return formattedTimestamp, nil
+	},
+	"io.debezium.time.MicroTimestamp": func(columnValue string, formatIfRequired bool) (string, error) {
+		epochMicroSecs, err := strconv.ParseInt(columnValue, 10, 64)
+		if err != nil {
+			return columnValue, fmt.Errorf("parsing epoch microseconds: %v", err)
+		}
+		epochSeconds := epochMicroSecs / 1000000
+		epochNanos := (epochMicroSecs % 1000000) * 1000
+		microTimeStamp := time.Unix(epochSeconds, epochNanos).UTC()
+		oracleTimestampFormat := "02-01-2006 03.04.05.000000 PM" //format: DD-MON-YYYY HH.MI.SS.FFFFFF PM
+		formattedTimestamp := microTimeStamp.Format(oracleTimestampFormat)
+		if formatIfRequired {
+			formattedTimestamp = fmt.Sprintf("TO_TIMESTAMP('%s','DD-MM-RR HH:MI:SS.FF9 AM')", formattedTimestamp)
+		}
+		return formattedTimestamp, nil
+	},
+	"io.debezium.time.NanoTimestamp": func(columnValue string, formatIfRequired bool) (string, error) {
+		epochNanoSecs, err := strconv.ParseInt(columnValue, 10, 64)
+		if err != nil {
+			return "", fmt.Errorf("parsing epoch nanoseconds: %v", err)
+		}
+		epochSeconds := epochNanoSecs / 1000000000
+		epochNanos := epochNanoSecs % 1000000000
+		nanoTimestamp := time.Unix(epochSeconds, epochNanos).UTC()
+		oracleTimestampFormat := "02-01-2006 03.04.05.000000000 PM" //format: DD-MON-YYYY HH.MI.SS.FFFFFFFFF PM
+		formattedTimestamp := nanoTimestamp.Format(oracleTimestampFormat)
+		if formatIfRequired {
+			formattedTimestamp = fmt.Sprintf("TO_TIMESTAMP('%s','DD-MM-RR HH:MI:SS.FF9 AM')", formattedTimestamp)
+		}
+		return formattedTimestamp, nil
+	},
+	"io.debezium.time.ZonedTimestamp": func(columnValue string, formatIfRequired bool) (string, error) {
+		debeziumFormat := "2006-01-02T15:04:05Z07:00"
+
+		parsedTime, err := time.Parse(debeziumFormat, columnValue)
+		if err != nil {
+			return "", fmt.Errorf("parsing timestamp: %v", err)
+		}
+
+		oracleFormat := "06-01-02 3:04:05.000000000 PM -07:00"
+		format := "RR-MM-DD HH:MI:SS.FF9 AM TZR"
+		timstampFn := "TO_TIMESTAMP_TZ"
+		if parsedTime.Location().String() == "UTC" { //LOCAL TIMEZONE Case
+			oracleFormat = "06-01-02 3:04:05.000000000 PM" // TODO: for timezone ones sqlldr is inserting as GMT though GMT/UTC is similar
```

Comment:
Hmm, I'm a little unsure about this. What is this scenario where we don't have to include timezone information while inserting? It's a TZ included field, so I think we should always make the timezone explicit.